### ISSUE TITLE
train on one gpu

### ISSUE THREAD

**Issue by 17670426157** [Lsan2401/RMSIN#1]
could you please let me know if you have the capability to conduct training on a single GPU?I attempted to modify the code for multi-GPU training but was unsuccessful.
![image](https://github.com/Lsan2401/RMSIN/assets/82247276/42b0fb63-a9c0-4137-b29e-efc9be2c6280)

**Comment by ciqian6** [comment]
There is a problem with loading data, I looked at the dataset and there is no such picture, but there is a tag with this name, and it still doesn't work if I delete the tag.
![image](https://github.com/Lsan2401/RMSIN/assets/82247276/4237675e-3b6f-4891-b538-ababcc9dafb5)

**Comment by Lsan2401** [comment]
> could you please let me know if you have the capability to conduct training on a single GPU?I attempted to modify the code for multi-GPU training but was unsuccessful. ![image](https://private-user-images.githubusercontent.com/82247276/294749468-42b0fb63-a9c0-4137-b29e-efc9be2c6280.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ2ODA0MDUsIm5iZiI6MTcwNDY4MDEwNSwicGF0aCI6Ii84MjI0NzI3Ni8yOTQ3NDk0NjgtNDJiMGZiNjMtYTljMC00MTM3LWIyOWUtZWZjOWJlMmM2MjgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDAyMTUwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZjBhZGMyM2YwZmNhMTYzMWY1MGZiN2RkODFmNTg1ZTFmZGUwMzBmYThmOWI0OTcxZmMzYWE4YmI2MWY4MGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M7VWrWCiCn5vSbx0gkM_A_KgviG1PaGoDi9fYyuoWrE)

The code is implemented to utilize DistributedDataParallel. If you intend to run it on a single GPU, you can either set the parameter `nproc_per_node` to 1 or comment out the corresponding code.

**Comment by ciqian6** [comment]
Thank you very much for your answer, I successfully ran on 1GPU, but it shows that my video memory is not enough, can you tell me how much graphics memory I need to run this code?

**Comment by Lsan2401** [comment]
> Thank you very much for your answer, I successfully ran on 1GPU, but it shows that my video memory is not enough, can you tell me how much graphics memory I need to run this code?

The Params of our model is 216M. You may consider decreasing the batch size or reducing the input image size, which may impact performance but can enable the model to run on your hardware.

**Comment by ciqian6** [comment]
The weights I trained can be evaluated for accuracy during training, but it doesn't work when I use the saved weights for a separate test. Can you help me with this?Thank you very much for your prompt response.

![image](https://github.com/Lsan2401/RMSIN/assets/82247276/8ab599c2-5356-4f99-8648-6b3ba3b1e92f)

**Comment by ciqian6** [comment]
Forgive me for bothering you multiple times. I tried modifying the code by incorporating the continuation of training part here, and now it runs. However, I'm unsure if there's an issue with my network. Could you provide me with a copy of the final trained weights if possible? Thank you very much if you can.
![image](https://github.com/Lsan2401/RMSIN/assets/82247276/249f2adc-1163-4b8d-95b5-4d5c4ee4f801)

**Comment by vvuonghn** [comment]
Hi @ciqian6 
The image https://github.com/Lsan2401/RMSIN/issues/1#issuecomment-1880039866 show your data does not exist. Please check

**Comment by Lsan2401** [comment]
> Forgive me for bothering you multiple times. I tried modifying the code by incorporating the continuation of training part here, and now it runs. However, I'm unsure if there's an issue with my network. Could you provide me with a copy of the final trained weights if possible? Thank you very much if you can. ![image](https://private-user-images.githubusercontent.com/82247276/296552701-249f2adc-1163-4b8d-95b5-4d5c4ee4f801.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1NDQ3MzgsIm5iZiI6MTcwNTU0NDQzOCwicGF0aCI6Ii84MjI0NzI3Ni8yOTY1NTI3MDEtMjQ5ZjJhZGMtMTE2My00YjhkLTk1YjUtNGQ1YzRlZTRmODAxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDAyMjAzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkNDFlYThjZWNhM2I1YjgwNzg2MDliYTRjN2E0MGFkZjM5MTk2OWJlYTc3YzU1OGE2NWM2MTg3ODg4Mjk1ZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xdbKXTqniRPLypGikFhRZID3e7Rp-862Xv-xjAAwy98)

It seems like the test dataset wasn't set correctly, resulting in the number of test data being shown as '[0 / 1]' and causing unexpected results. Please check the dataset path. And we will be releasing our weights soon.

**Comment by Inverse-function** [comment]
> 在训练期间，可以评估我训练的权重的准确性，但是当我将保存的权重用于单独的测试时，它不起作用。你能帮我解决这个问题吗？非常感谢您的及时回复。
> 
> ![图像](https://private-user-images.githubusercontent.com/82247276/296551838-8ab599c2-5356-4f99-8648-6b3ba3b1e92f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxNzQ4MzMsIm5iZiI6MTcxNjE3NDUzMywicGF0aCI6Ii84MjI0NzI3Ni8yOTY1NTE4MzgtOGFiNTk5YzItNTM1Ni00Zjk5LTg2NDgtNmIzYmEzYjFlOTJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIwVDAzMDg1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5NmFkMjk5ZmExNmQ3ZTU3OTdkNmE1MGRjMjg2ZjI5YTI2NmQ2MDcyYmQwYTU3NjgyMzk4YjkyZWNkMWQzYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JoPtVV40rmJLUTubV3nyi4cMpai9LfRlmhKQ09WigJg)

你好，我能问一下这个问题你是怎么解决的吗？我也发生了同样的问题。

**Comment by ciqian6** [comment]
@Inverse-function 我咋看不到你的图片，是测试的精度比训练的较低吗？

**Comment by Inverse-function** [comment]
> @Inverse-function 我咋看不到你的图片，是测试的精度比训练的较低吗？

不不不，是训练完加载checkpoint测试的时候，显示尺寸不匹配

**Comment by ciqian6** [comment]
那就把加载权重的代码换成他训练的那部分代码试试 @Inverse-function

**Comment by Inverse-function** [comment]
> 那就把加载权重的代码换成他训练的那部分代码试试 @Inverse-function

抱歉，我没太理解什么意思
CUDA_VISIBLE_DEVICES=0,1,2,3 python -m torch.distributed.launch --nproc_per_node 4 --master_port 12345 train.py --dataset rrsisd --model_id RMSIN --batch-size 8 --lr 0.00005 --wd 1e-2 --swin_type base --pretrained_swin_weights ./pretrained_weights/swin_base_patch4_window12_384_22k.pth --epochs 40 --img_size 480 2>&1 | tee ./output

python test.py --swin_type base --dataset rrsisd --resume ./your_checkpoints_path --split val --workers 4 --window12 --img_size 480
上边是训练的，下边是测试的，你的意思是把上边的train.py改成test.py吗

**Comment by ciqian6** [comment]
@Inverse-function 不是改调用测试的命令，是改test.py文件的代码

**Comment by JackHeroFly** [comment]
> Thank you very much for your answer, I successfully ran on 1GPU, but it shows that my video memory is not enough, can you tell me how much graphics memory I need to run this code?
Hello，can you tell me that how you ran on 1 GPU? thank you very much.
